quinn-proto/src/crypto/ring.rs

```diff
@@ -70,3 +70,42 @@ impl crypto::HmacKey for hmac::Key {
         hmac::verify(self, data, signature).map_err(|_| ())
     }
 }
+
+impl crypto::PseudoRandomKey for hkdf::Prk {
+    fn from_secret(bytes: &[u8]) -> Self {
+        let salt = hkdf::Salt::new(hkdf::HKDF_SHA256, &[]);
+        salt.extract(bytes)
+    }
+
+    fn expand(&self, random_bytes: &[u8], mut key_buffer: &mut [u8]) {
+        let info = [random_bytes];
+        let okm = self.expand(&info, hkdf::HKDF_SHA256).unwrap();
```

Comment:
I think we try to avoid unwrapping in nested library code like this. Return a `Result` instead (potentially just `Result<(), ()>`), which can be propagated outward until it reaches a level where the caller knows how to deal with it.